posthog-core/src/index.ts

```diff
@@ -443,6 +444,19 @@ export abstract class PostHogCore {
           }
           this.setKnownFeatureFlags(newFeatureFlags)
         }
+
+        if (res.featureFlagPayloads) {
+          let newFeatureFlagPayloads = res.featureFlagPayloads
+          if (res.errorsWhileComputingFlags) {
+            // if not all flags were computed, we upsert flags instead of replacing them
+            const currentFlagPayloads = this.getPersistedProperty<PostHogDecideResponse['featureFlagPayloads']>(
+              PostHogPersistedProperty.FeatureFlagPayloads
+            )
+            newFeatureFlagPayloads = { ...currentFlagPayloads, ...res.featureFlagPayloads }
+          }
+          this.setKnownFeatureFlagPayloads(newFeatureFlagPayloads)
+        }
```

Comment:
Do we need upserting here

Reply:
Yeah. Actually, can we not duplicate upserting code and move it into the above? wherever flags are set, set the payloads as well, otherwise don't do anything.

Let's always keep flags response as the source of truth, and payloads just follow whatever happens to flags.

Reply:
Otherwise, will be easy to make these go out of sync, which is terrible :/